### ISSUE TITLE
Could not install memcached-3.1.5 on PHP 7.4.15

### ISSUE THREAD

**Issue by ruudk** [shivammathur/setup-php#423]
**Describe the bug**
memcached fails installing since today.

**Version**
2.10.0

**Runners**

- [x] GitHub Hosted
- [ ] Self Hosted

**Operating systems**
Linux

**PHP versions**
7.4

**To Reproduce**
```yaml
            -   name: Setup PHP
                uses: shivammathur/setup-php@2.10.0
                with:
                    php-version: 7.4
                    tools: cs2pr
                    extensions: mbstring,bcmath,intl,gd,zip,memcached-3.1.5,redis-5.3.3,rdkafka-5.0.0
                env:
                    fail-fast: true
```

**Expected behavior**
This suddenly broke today. 

**Screenshots/Logs**
```

==> Setup PHP
✓ PHP Switched to PHP 7.4.15

==> Setup Tools
✓ composer Added composer 2.0.11
✓ cs2pr Added cs2pr 1.4.0

==> Setup Extensions
✓ mbstring Enabled
✓ bcmath Enabled
✓ intl Enabled
✓ gd Enabled
✓ zip Enabled
✗ memcached Could not install memcached-3.1.5 on PHP 7.4.15
```

**Comment by ruudk** [comment]
When I install `memcached` instead of `memcached-3.1.5` it works.

**Comment by shivammathur** [comment]
@ruudk 

Yes `memcached` package is `3.1.5`. So you can just specify `memcached` without version and it should work. 

```yml
- name: Setup PHP
  uses: shivammathur/setup-php@v2
  with:
    php-version: 7.4
    extensions: memcached
```

Test workflow: https://github.com/shivammathur/test-setup-php/runs/2021730562?check_suite_focus=true

If you want to compile using pecl, you will have to install `libmemcached-dev` for memcached and `librdkafka-dev` for rdkafka before running `setup-php`.

```yml
- name: Install required libraries 
  run: sudo apt-get install -y libmemcached-dev librdkafka-dev
- name: Setup PHP
  uses: shivammathur/setup-php@verbose
  with:
      php-version: 7.4
      tools: cs2pr
      extensions: mbstring,bcmath,intl,gd,zip,memcached-3.1.5,redis-5.3.3,rdkafka-5.0.0
```

Test workflow: https://github.com/shivammathur/test-setup-php/actions/runs/617443514/workflow

**Comment by ruudk** [comment]
But it's weird that I had this config for a very long time without issues, and today it broke all of a sudden. When you specify the version, will it then always install from source?

**Comment by shivammathur** [comment]
Since memcached is pre-installed and the input version matches like right now it both `3.1.5` in package and in input, It is just enabled, when they differ the extension is installed from source using pecl.

**Comment by ruudk** [comment]
So this is the rule:
- When no version provided, use default installed version (no pecl install)
- When versiokn is provided and matches the default installed version, just enable it (no pecl install)
- When version provided, install from PECL when the version does not match the default installed version

?

**Comment by shivammathur** [comment]
Yes

**Comment by ruudk** [comment]
Then how come it broke today? Because I provide `memcached-3.1.5` and that's the default installed version, but I still get: `✗ memcached Could not install memcached-3.1.5 on PHP 7.4.15`

**Comment by shivammathur** [comment]
Can you please give me link to the logs where it is failing.

**Comment by shivammathur** [comment]
I'm not able to reproduce the issue with PHP 7.4: https://github.com/shivammathur/test-setup-php/actions/runs/617493608/workflow

**Comment by ruudk** [comment]
Created a PR with my config, but it does not start the workflow: https://github.com/shivammathur/test-setup-php/pull/3

**Comment by alexsegura** [comment]
The same happens with Redis

```yaml
- name: Setup PHP
   uses: shivammathur/setup-php@v2
   with:
     php-version: '7.4'
     tools: composer:v2, phpstan, redis
```

```
==> Setup PHP
✓ PHP Switched to PHP 7.4.15

==> Setup Tools
✓ composer Added composer 2.0.11
✓ phpstan Added phpstan 0.12.80
✗ redis Tool redis is not supported
```

**Comment by ruudk** [comment]
should be extension, no?

**Comment by alexsegura** [comment]
@ruudk well it was working until today 🤷‍♂️

**Comment by ruudk** [comment]
Ok, glad to hear I'm not crazy 😅 Something _did change_ today

**Comment by alexsegura** [comment]
I see some changes about Redis & other things between [2.9.0 and 2.10.0](https://github.com/shivammathur/setup-php/compare/2.9.0...2.10.0)

What is strange is that `2.10.0` was released 8 days ago... maybe there's some cache on GitHub Actions, and it was actually invalidated today?

Probably, this will work:

```yaml
uses: shivammathur/setup-php@2.9.0
```

**Comment by ruudk** [comment]
@alexsegura See this comment https://github.com/shivammathur/test-setup-php/pull/3#issuecomment-789692159

**Comment by shivammathur** [comment]
@alexsegura redis is not a tool, it is an extension. Please create a separate issue and provide your workflow. I will have a look.

**Comment by alexsegura** [comment]
@shivammathur like I said, it used to work just like that. 
Don't worry, I will wait until you fix this issue, and fix my workflow accordingly.

**Comment by shivammathur** [comment]
@ruudk 
The issue was in ubuntu-20.04 image, which was deployed yesterday. I was able to ssh into it and memcached was not installed in the build for `PHP 7.4`.

if you install `libmemcached-dev` above setup-php when specifying the memcached version. It will install memached successfully, or you can just specify memcached without version and  it will install from the package.

Both of these should work


```yml
- name: Install required libraries 
  run: sudo apt-get install -y libmemcached-dev librdkafka-dev
- name: Setup PHP
  uses: shivammathur/setup-php@v2
  with:
      php-version: 7.4
      extensions: memcached-3.1.5
```

```yml
- name: Setup PHP
  uses: shivammathur/setup-php@v2
  with:
    php-version: 7.4
    extensions: memcached
```

**Comment by ruudk** [comment]
Thanks for debugging this. 

I will add `libmemcached-dev`. But shouldn't `setup-php` detect that `memcached-3.1.5` is included already and just enable that instead? Without failure.

**Comment by shivammathur** [comment]
Yes, that can be improved in future release here. I will make a note of it.

This issue was caused by change in how `ondrej/ppa` packages are built.
More details here.
https://www.patreon.com/posts/47793617

I will create a pull request to the actions images to get this fixed.

**Comment by shivammathur** [comment]
Should be fixed, once this is merged and deployed
https://github.com/actions/virtual-environments/pull/2834

**Comment by ruudk** [comment]
@shivammathur Now that https://github.com/actions/virtual-environments/pull/2834 is merged, should it work immediately?

**Comment by shivammathur** [comment]
@ruudk No, it will have to deployed first. There will be a release on [actions/virtual-environment](https://github.com/actions/virtual-environments/tags) Ubuntu 20.04 20210304.2 or newer when they start deploying it. Till then please use https://github.com/shivammathur/setup-php/issues/423#issuecomment-789708606

**Comment by shivammathur** [comment]
@ruudk Fix has been deployed and `memcached-3.1.5` should work now.
